modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/SnapshotAwarePartitionDataStorage.java

```diff
@@ -123,11 +127,21 @@ private void handleSnapshotInterference(RowId rowId) {
 
     @Override
     public void close() throws Exception {
-        // TODO: IGNITE-17935 - terminate all snapshots of this partition considering correct locking to do it consistently
+        cleanupSnapshots();
 
         partitionStorage.close();
     }
 
+    private void cleanupSnapshots() {
+        PartitionSnapshots partitionSnapshots = getPartitionSnapshots();
+
+        try (AutoLockup ignored = partitionSnapshots.acquireReadLock()) {
```

Comment:
Let's change the name **ignored**.

Reply:
Maybe `readLock` ?

Reply:
Renamed it to `ignoredLockup` (because it is a lockup and not a lock, and because it is ignored; naming it with 'ignored' in the beginning makes IDEA stop nagging about an 'unused variable')

Reply:
I don’t understand a bit, the variable is called “not used”, but in fact it is used, it does not make it clear, reading the code I think if it is not used then the code is not needed and would have been deleted.

Reply:
The **object** is used by try-with-resources machinery, but the **variable** is not used inside `try` block. 

Try to write something like the following in IDEA:

```
try (var v = new FileOutputStream(...)) {
    ... some code
}
```

IDEA will immediately nag that `v` is not used. I want to avoid this behavior.

Reply:
In the above code, the variable is really not used, and we can say that I can safely remove this block.
In your case, the variable is not used and the block is painful to remove.